x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.test.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { FlashMessagesLogic, IFlashMessage } from './flash_messages_logic';
+
+describe('FlashMessagesLogic', () => {
+  const DEFAULT_VALUES = {
+    messages: [],
+    queuedMessages: [],
+    historyListener: null,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    resetContext({});
+  });
+
+  it('has expected default values', () => {
+    FlashMessagesLogic.mount();
+    expect(FlashMessagesLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('setMessages()', () => {
+    it('sets an array of messages', () => {
+      const messages = [
+        { type: 'success', message: 'Hello world!!' },
+        { type: 'error', message: 'Whoa nelly!', description: 'Uh oh' },
+        { type: 'info', message: 'Everything is fine, nothing is ruined' },
+      ] as IFlashMessage[];
+
+      FlashMessagesLogic.mount();
+      FlashMessagesLogic.actions.setMessages(messages);
+
+      expect(FlashMessagesLogic.values.messages).toEqual(messages);
+    });
+
+    it('automatically converts to an array if a single message obj is passed in', () => {
+      const message = { type: 'success', message: 'I turn into an array!' } as IFlashMessage;
+
+      FlashMessagesLogic.mount();
+      FlashMessagesLogic.actions.setMessages(message);
```

Comment:
What happens if you call this twice, does it replace this previously set messages?

Reply:
It does yeah. I contemplated adding an `addMessage()` fn that simply pushes / doesn't completely wipe out previous messages, but we haven't really seen a need for that so far - we definitely could in the future though if that use case popped up